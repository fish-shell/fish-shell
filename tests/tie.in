# Test tied variables.

echo '# At this point we should have just the default tied vars.'
tie

echo '# Add a tied var.'
tie XYZ
tie

echo '# Add a custom tied var using period as the separator.'
tie -s . ABC abc_array
tie

echo '# Add a custom tied var using space as the separator.'
tie -s ' ' SPACE
tie

echo '# Test modifying the array var by setting it to zero elements.'
set xyz
echo $xyz
echo $XYZ

echo '# Test modifying the array var by setting it to one element.'
set xyz one-string
echo $xyz
echo $XYZ

echo '# Test modifying the array var by setting it to more than one element.'
set xyz abc def ghi
echo $xyz
echo $XYZ

echo '# Test modifying the scalar var by setting it to zero elements.'
set XYZ
echo $xyz
echo $XYZ

echo '# Test modifying the scalar var by setting it to one element.'
set XYZ ONE-STRING
echo $xyz
echo $XYZ


echo '# Test modifying the scalar var by setting it to more than one element.'
set XYZ a:b:c:d
echo $xyz
echo $XYZ

echo '# Test modifying the scalar var with a non-standard separator.'
set ABC a:b.c:d.e.f
echo $abc_array
echo $ABC

echo '# Test modifying the array var with a non-standard separator.'
set abc_array aa:bb $abc_array yy:zz
echo $abc_array
echo $ABC

echo '# Removing a tie should leave the tied vars unchanged and no longer kept in sync.'
tie -U ABC
echo $abc_array
echo $ABC
set abc_array 1 2
set ABC 1:2:3
echo $abc_array
echo $ABC

echo '# Test modifying the tied var with a space separator.'
set space s p a c e
printf "space |%s|\n" $space
printf "SPACE |%s|\n" $SPACE
echo count space (count $space)
echo count SPACE (count $SPACE)
set SPACE 's:p a c:e'
printf "space |%s|\n" $space
printf "SPACE |%s|\n" $SPACE
echo count space (count $space)
echo count SPACE (count $SPACE)

echo '# Changing a tied var in a nested block should affect the global scope.'
begin
    set xyz 1 3 5
end
echo $xyz
echo $XYZ

echo '# Changing a tied var in a nested block via `set -l` should not affect the global scope.'
set -g not_tied xx yy
begin
    set -l not_tied abc def
    set -l XYZ $XYZ
    set -l xyz 2 4 6
    echo $not_tied
    echo $xyz
    echo $XYZ
end
echo $not_tied
echo $xyz
echo $XYZ

echo '# Erasing a tied var should erase it\'s mirror but should leave the tie in place.'
set -e XYZ
set -q XYZ
and echo XYZ unexpectedly defined
set -q xyz
and echo xyz unexpectedly defined
set XYZ salmon:tuna:cod
echo $xyz
echo $XYZ
set xyz clam mollusk oyster
echo $xyz
echo $XYZ
