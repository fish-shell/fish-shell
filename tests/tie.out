# At this point we should have just the default tied vars.
tie --sep : CDPATH fish_cdpath
tie --sep : MANPATH fish_manpath
tie --sep : PATH fish_path
# Add a tied var.
tie --sep : CDPATH fish_cdpath
tie --sep : MANPATH fish_manpath
tie --sep : PATH fish_path
tie --sep : XYZ xyz
# Add a custom tied var using period as the separator.
tie --sep . ABC abc_array
tie --sep : CDPATH fish_cdpath
tie --sep : MANPATH fish_manpath
tie --sep : PATH fish_path
tie --sep : XYZ xyz
# Add a custom tied var using space as the separator.
tie --sep . ABC abc_array
tie --sep : CDPATH fish_cdpath
tie --sep : MANPATH fish_manpath
tie --sep : PATH fish_path
tie --sep ' ' SPACE space
tie --sep : XYZ xyz
# Test modifying the array var by setting it to zero elements.


# Test modifying the array var by setting it to one element.
one-string
one-string
# Test modifying the array var by setting it to more than one element.
abc def ghi
abc:def:ghi
# Test modifying the scalar var by setting it to zero elements.


# Test modifying the scalar var by setting it to one element.
ONE-STRING
ONE-STRING
# Test modifying the scalar var by setting it to more than one element.
a b c d
a:b:c:d
# Test modifying the scalar var with a non-standard separator.
a:b c:d e f
a:b.c:d.e.f
# Test modifying the array var with a non-standard separator.
aa:bb a:b c:d e f yy:zz
aa:bb.a:b.c:d.e.f.yy:zz
# Removing a tie should leave the tied vars unchanged and no longer kept in sync.
aa:bb a:b c:d e f yy:zz
aa:bb.a:b.c:d.e.f.yy:zz
1 2
1:2:3
# Test modifying the tied var with a space separator.
space |s|
space |p|
space |a|
space |c|
space |e|
SPACE |s p a c e|
count space 5
count SPACE 1
space |s:p|
space |a|
space |c:e|
SPACE |s:p a c:e|
count space 3
count SPACE 1
# Changing a tied var in a nested block should affect the global scope.
1 3 5
1:3:5
x1 y3 z5
x1:y3:z5
# Changing a tied var in a nested block via `set -l` should not affect the global scope.
abc def
2 4 6
2:4:6
g h i
g-h-i
xx yy
x1 y3 z5
x1:y3:z5
ghi not set in outer scope as expected
GHI not set in outer scope as expected
# Erasing a tied var should erase it's mirror but should leave the tie in place.
salmon tuna cod
salmon:tuna:cod
clam mollusk oyster
clam:mollusk:oyster
